src/tables/cbdt.rs

```diff
@@ -52,28 +47,72 @@ impl<'a> Table<'a> {
                 s.skip::<i8>(); // ver_bearing_y
                 s.skip::<u8>(); // ver_advance
-                let data_len = s.read::<u32>()?;
+                Metrics {
+                    x: hor_bearing_x,
+                    y: hor_bearing_y,
+                    width,
+                    height,
+                }
+            }
+            MetricsFormat::Shared => {
+                location.metrics
+            }
+        };
+        match location.format.data {
+            BitmapDataFormat::ByteAligned { bit_depth } => {
+                let row_len = (u32::from(metrics.width) * u32::from(bit_depth) + 7) / 8;
+                let data_len = row_len * u32::from(metrics.height);
                 let data = s.read_bytes(usize::num_from(data_len))?;
                 Some(RasterGlyphImage {
-                    x: i16::from(hor_bearing_x),
+                    x: i16::from(metrics.x),
                     // `y` in CBDT is a bottom bound, not top one.
-                    y: i16::from(hor_bearing_y) - i16::from(height),
-                    width: u16::from(width),
-                    height: u16::from(height),
+                    y: i16::from(metrics.y) - i16::from(metrics.height),
+                    width: u16::from(metrics.width),
+                    height: u16::from(metrics.height),
                     pixels_per_em: location.ppem,
-                    format: RasterImageFormat::PNG,
+                    format: match bit_depth {
+                        1 => RasterImageFormat::BitmapMono,
+                        2 => RasterImageFormat::BitmapGray2,
+                        4 => RasterImageFormat::BitmapGray4,
+                        8 => RasterImageFormat::BitmapGray8,
+                        32 => RasterImageFormat::BitmapBgra32,
+                        _ => return None,
+                    },
+                    data,
+                })
+            }
+            BitmapDataFormat::BitAligned { bit_depth } => {
+                let data_len = (u32::from(metrics.width) 
+                    * u32::from(metrics.height)
+                    * u32::from(bit_depth) + 7) / 8;
+                let data = s.read_bytes(usize::num_from(data_len))?;
+                Some(RasterGlyphImage {
+                    x: i16::from(metrics.x),
+                    // `y` in CBDT is a bottom bound, not top one.
+                    y: i16::from(metrics.y) - i16::from(metrics.height),
+                    width: u16::from(metrics.width),
+                    height: u16::from(metrics.height),
+                    pixels_per_em: location.ppem,
+                    format: match bit_depth {
+                        1 => RasterImageFormat::BitmapMonoPacked,
+                        2 => RasterImageFormat::BitmapGray2Packed,
+                        4 => RasterImageFormat::BitmapGray4Packed,
+                        8 => RasterImageFormat::BitmapGray8,
+                        32 => RasterImageFormat::BitmapBgra32,
```

Comment:
Is it defined anywhere in the spec? I cannot find it. In the MS/OpenType spec, 32 isn't allowed. Is it OTB-specific?

Reply:
This is defined in the [CBDT ](https://learn.microsoft.com/en-us/typography/opentype/spec/cbdt) specs, under "Uncompressed Color Bitmaps"

Reply:
Oh, so it's not from EBDT. Got it.

Also I would call it `BitmapPremulBgra32`

Reply:
Updated